src/_macosx.m

```diff
@@ -1093,38 +1002,21 @@ -(void)save_figure:(id)sender
 static int
 NavigationToolbar2_init(NavigationToolbar2 *self, PyObject *args, PyObject *kwds)
 {
-    PyObject* obj;
-    FigureCanvas* canvas;
+    PyObject* canvas;
     View* view;
-
-    int i;
     NSRect rect;
-    NSSize size;
     NSSize scale;
-
     const float gap = 2;
     const int height = 36;
     const int imagesize = 24;
 
     self->height = height;
 
-    obj = PyObject_GetAttrString((PyObject*)self, "canvas");
-    if (obj==NULL)
-    {
-        PyErr_SetString(PyExc_AttributeError, "Attempt to install toolbar for NULL canvas");
-        return -1;
-    }
-    Py_DECREF(obj); /* Don't increase the reference count */
-    if (!PyObject_IsInstance(obj, (PyObject*) &FigureCanvasType))
-    {
-        PyErr_SetString(PyExc_TypeError, "Attempt to install toolbar for object that is not a FigureCanvas");
+    if (!(canvas = PyObject_GetAttrString((PyObject*)self, "canvas"))) {
```

Comment:
Does this actually set an error (it used to)?

Reply:
PyObject_GetAttrString sets an error if the attribute is not there; FigureCanvas_view sets one if view is NULL (or if self is not a FigureCanvas).
However I realized it's even easier to just pass the canvas as argument to the constructor...